poseidon2-starky/src/generation.rs

```diff
@@ -1,16 +1,37 @@
-use crate::columns::{COL_INPUT_START, COL_OUTPUT_START, NUM_COLS, STATE_SIZE};
+use crate::columns::{
+    COL_1ST_FULLROUND_STATE_START, COL_2ND_FULLROUND_STATE_START, COL_INPUT_START,
+    COL_OUTPUT_START, COL_PARTIAL_ROUND_END_STATE_START, COL_PARTIAL_ROUND_STATE_START, NUM_COLS,
+    ROUNDS_F, ROUNDS_P, STATE_SIZE,
+};
+use ark_ff::{BigInteger, PrimeField};
 use num::bigint::BigUint;
 use plonky2::hash::hash_types::RichField;
 use std::convert::TryInto;
-use zkhash::ark_ff::{BigInteger, PrimeField};
+use std::ops::AddAssign;
 use zkhash::fields::goldilocks::FpGoldiLocks;
 use zkhash::poseidon2::poseidon2::Poseidon2;
 use zkhash::poseidon2::poseidon2_instance_goldilocks::POSEIDON2_GOLDILOCKS_8_PARAMS;
 
-/// Represent a row of the preimage
+pub(crate) fn scalar_to_field<RF: RichField, PF: PrimeField>(scalar: &PF) -> RF {
+    RF::from_noncanonical_biguint(BigUint::from_bytes_le(&scalar.into_bigint().to_bytes_le()))
```

Comment:
curious why PrimeField can't go to BigUint directly and have to go through BigInt?

Reply:
Oh I see these are two different crates ark_ff and num